internal/tester/data.go

```diff
@@ -292,6 +298,31 @@ func (t *DataTester) StartPeriodicLogger(
 	return ctx.Err()
 }
 
+// WatchEndCondition starts go routines to watch the end conditions
+func (t *DataTester) WatchEndCondition(
+	ctx context.Context,
+	config *configuration.Configuration,
+) error {
+	if config.Data.EndAtTip {
+		// runs a go routine that ends when reaching tip
+		go t.syncer.EndAtTipLoop(ctx, config.TipDelay, EndAtTipCheckInterval)
+	}
+
+	if config.Data.EndDuration != "" {
+		dur, err := time.ParseDuration(config.Data.EndDuration)
```

Comment:
We should mandate all durations be in seconds as a `uint64` (see HTTP timeout). I think it is important to be consistent about how we specify time in the config:
```
// HTTPTimeout is the timeout for HTTP requests in seconds.
HTTPTimeout uint64 `json:"http_timeout"`		HTTPTimeout uint64 `json:"http_timeout"
```

Reply:
Curious to hear other opinions, I guess end duration will typically be much longer than HTTP timeout.

Reply:
Definitely agree consistency is the key here 👍